status-group-chats-spec.md

```diff
@@ -0,0 +1,202 @@
+# Status Group Chat Specification
+
+> Version: 0.1 (Draft)
+>
+> Authors: Andrea Maria Piana <andreap@status.im>
+>
+
+
+## Table of Contents
+
+- [Abstract](#abstract)
+- [Membership updates](#membership-updates)
+  - [Chat ID](#chat-id)
+  - [Signature](#signature)
+  - [Group membership event](#group-membership-event)
+    - [chat-created](#chat-created)
+    - [name-changed](#name-changed)
+    - [members-added](#members-added)
+    - [members-joined](#members-joined)
+    - [admins-added](#admins-added)
+    - [members-removed](#members-removed)
+    - [admin-removed](#admin-removed)
+
+
+## Abstract
+
+This documents describes the group chat protocol used by the status application. Pairwise encryption is used among member so a message is exchanged between each participants, similarly to a one-to-one message.
+
+## Membership updates
+
+Membership updates messages are used to propagate group chat membership changes. The transit format is described in the [Status Payload Specs](status-payload-specs.md). Here we will be describing each specific field.
+
+The format is:
+
+```
+{
+  "events": [],
+  "signature": string,
+  "chat-id": string
+}
+```
+
+### Chat ID
+
+Each membership update MUST be sent with a corresponding `chat-id`. 
+The format of this chat id MUST be a string, [UUID](https://tools.ietf.org/html/rfc4122 ), concatenated with the hex-encoded public key of the creator of the chat. This chat-id MUST be validated by all clients, and should be discarded if it does not follow these rules.
+
+### Signature
+
+The signature for each event is calculated by creating a JSON array of all the `events` sorted by `clock-value` in ascending order, where each event is transformed in an array of tuples `field-name`, `value`. The last element of the array MUST be the `chat-id`. 
+
+For example the event:
+
+```
+   {
+     "chat-id": "chat-id",
+       "events":  [
+       {"b": "b-value"
+        "clock-value": 1,
+        "a": "a-value"
+        },
+       {
+         "e": "e-value",
+         "clock-value": 0,
+         "a": "a-value"
+       }
+       ]
+   }
+
+```
+
+Results in the structure:
+
+```
```

Comment:
What about the order of fields? It's undefined in JSON objects. If we transform it in this way, we need to impose some order so that all clients do it in the same way.

Reply:
we sort them alphabetically ascending, forgot to add that to the description